infrastructure/template.yaml

```diff
@@ -701,20 +704,27 @@ Resources:
             Resource: "*"
             Action: "kms:*"
             Principal:
-              AWS: !Sub "arn:aws:iam::${AWS::AccountId}:role/aws-reserved/sso.amazonaws.com/eu-west-2/AWSReservedSSO_di-ipv-cri-otg-hmrc-prod-adm-kms_c1de3f0ff8700ff0"
+              AWS: !Sub arn:aws:iam::${AWS::AccountId}:root
+            Condition:
+              StringLike:
+                "aws:PrincipalArn":
+                  - !Sub "arn:aws:iam::${AWS::AccountId}:role/aws-reserved/sso.amazonaws.com/${AWS::Region}/AWSReservedSSO_ApprovedAdmin*"
+                  - !Sub "arn:aws:iam::${AWS::AccountId}:role/aws-reserved/sso.amazonaws.com/${AWS::Region}/AWSReservedSSO_AdministratorAccessPermission*"
+
```

Comment:
I actually think we should be able to remove the allow and just let the deny handle it

Reply:
Could for the RBP but not the key policy as this needs explicit allow https://docs.aws.amazon.com/kms/latest/developerguide/key-policies.html